src/flash/flasher.cpp

```diff
@@ -0,0 +1,151 @@
+#include "logger.h"
+#include "flasher.h"
+
+#include <QCoreApplication>
+#include <QDebug>
+#include <QFileInfo>
+#include <QProcess>
+#include <QRegularExpression>
+#include <QSerialPortInfo>
+#include <QThread>
+
+PING_LOGGING_CATEGORY(FLASH, "ping.flash")
+
+Flasher::Flasher(QObject* parent)
+    : QObject(parent)
+{
+    _binRelativePath =
+#ifdef Q_OS_OSX
+        // macdeployqt file do not put stm32flash binary in the same folder of pingviewer
+        QCoreApplication::applicationDirPath() + "/../..";
+#else
+        QCoreApplication::applicationDirPath();
+#endif
+}
+
+bool Flasher::setBaudRate(int baudRate)
+{
+    if(!_validBaudRates.contains(baudRate)) {
+        qCCritical(FLASH) << "Invalid baud rate:" << baudRate;
+        qCCritical(FLASH) << "Valid baud rates are:" << _validBaudRates;
+        return false;
+    }
+
+    _baudRate = baudRate;
+    return true;
+};
+
+bool Flasher::setFirmwarePath(const QString& firmwareFilePath)
+{
+    QFileInfo fileInfo(firmwareFilePath);
+    if(!fileInfo.exists()) {
+        qCCritical(FLASH) << "Firmware file does not exist:" << firmwareFilePath;
+        return false;
+    }
+
+    _firmwareFilePath = fileInfo.absoluteFilePath();
+    return true;
+};
+
+bool Flasher::setLink(const LinkConfiguration& link)
+{
+    if(!link.checkType(LinkType::Serial)) {
+        qCCritical(FLASH) << "Link configuration is not valid:" << link;
+        return false;
+    }
+    _link = link;
+    return true;
+};
+
+void Flasher::setVerify(bool verify)
+{
+    _verify = verify;
+};
+
+QString Flasher::stm32flashPath()
+{
+#ifdef Q_OS_OSX
+    // macdeployqt file do not put stm32flash binary in the same folder of pingviewer
+    static QString binPath = QCoreApplication::applicationDirPath() + "/../..";
+#else
+    static QString binPath = QCoreApplication::applicationDirPath();
+#endif
+#ifdef Q_OS_WIN
+    return binPath + "/stm32flash.exe";
```

Comment:
`_binRelativePath`?

Reply:
Oy, we're not using this https://github.com/bluerobotics/ping-viewer/pull/361/files#diff-21335bea06bc2e8e55ecdd147181ff74R17

It's redundant here, no?